### PR TITLE
Add large capital opportunity search app

### PR DESCRIPTION
**Pull request by elcct**
### Description of change

A new `POST /v4/search/large-capital-opportunity` endpoint has been added.
The endpoint enables the following filters:

Main filters:

    type
    status
    name
    created_by

Detail filters:

    uk_region_location
    promoter
    promoter_name
    lead_dit_relationship_manager
    required_checks_conducted
    required_checks_conducted_by
    asset_class
    opportunity_value_type
    opportunity_value_start
    opportunity_value_end
    construction_risk

Requirement filters:

    total_investment_sought_start
    total_investment_sought_end
    current_investment_secured_start
    current_investment_secured_end
    investment_type
    estimated_return_rate
    time_horizon

Extra filters:

    created_on_after
    created_on_before

### Checklist

* [x] If this is a releasable change, has a news fragment been added?

  <details>
  <summary>Explanation</summary>
  
  A news fragment is required for any releasable change (i.e. code that runs in or affects production) so that a corresponding changelog entry is added when releasing.
  
  Check [changelog/README.md](https://github.com/uktrade/data-hub-api/blob/master/changelog/README.md) for instructions.
  
  </details>
  
* [x] Has this branch been rebased on top of the current `develop` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [x] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/develop/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: datahub/search/large_capital_opportunity/test/test_signals.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    es_with_signals.indices.refresh()
+    assert _get_es_document(es_with_signals, opportunity.pk)
+
+
+def test_updated_large_capital_opportunity_synced(es_with_signals):
+    """Test that when a large capital opportunity is updated it is synced to ES."""
+    opportunity = LargeCapitalOpportunityFactory()
+    opportunity.total_investment_sought = 12345
+    opportunity.save()
+    es_with_signals.indices.refresh()
```
Comment (tomkingston) [inline datahub/search/large_capital_opportunity/test/test_signals.py:32]: There doesn't seem to be an assertion here - presumably you want to check that the es object has the new value for `total_investment_sought`?

File: datahub/search/large_capital_opportunity/test/test_signals.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    ),
+)
+def test_delete_from_es(
+    opportunity_factory,
+    expected_in_index,
+    expected_to_call_delete,
+    es_with_signals,
+):
+    """
+    Test that when a large capital opportunity is deleted from db it is also
```
Comment (tomkingston) [inline datahub/search/large_capital_opportunity/test/test_signals.py:48]: *it is also => it also

File: datahub/search/large_capital_opportunity/test/test_signals.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    opportunity.save()
+    es_with_signals.indices.refresh()
+
+
+@pytest.mark.parametrize(
+    'opportunity_factory,expected_in_index,expected_to_call_delete',
+    (
+        (LargeCapitalOpportunityFactory, True, True),
+    ),
+)
```
Comment (tomkingston) [inline datahub/search/large_capital_opportunity/test/test_signals.py:40]: Is there a lot of point of using parametrize with only one scenario? `expected_to_call_delete` is not used and the else from `if expected_in_index` never gets run.

**Review by tomkingston — COMMENTED** [review]
Just a couple of queries on the signal tests, but otherwise looks good :+1:

File: datahub/search/large_capital_opportunity/test/test_signals.py (hunk 1 above)
Comment (elcct) [reply datahub/search/large_capital_opportunity/test/test_signals.py:31]: Yes, correct. I added it now. Thank you!

File: datahub/search/large_capital_opportunity/test/test_signals.py (hunk 2 above)
Comment (elcct) [reply datahub/search/large_capital_opportunity/test/test_signals.py]: Nice one. Thank you!

File: datahub/search/large_capital_opportunity/test/test_signals.py (hunk 3 above)
Comment (elcct) [reply datahub/search/large_capital_opportunity/test/test_signals.py]: I copied this test from the investor profiles app and forgot to update it. I have removed parametrisation and unused paths. Thank you!

**Review by tomkingston — APPROVED** [review]

File: datahub/search/large_capital_opportunity/models.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
+from elasticsearch_dsl import Boolean, Date, Keyword, Long, Text
+
+from datahub.search import dict_utils
+from datahub.search import fields
```
Comment (ambrozic) [inline datahub/search/large_capital_opportunity/models.py:4]: this two imports could be inline

**Pull request merged**